Syntax error with `?` in any

## Bug Report

### 1. Minimal reproduce step (Required)

```kcl
_a = { zz = [ { type = "Complete", status ="True" } ]}

# broken
b = any c in _a?.zz {
    c.type == "Complete" and c.status == "True"
} 

# works
d = any c in _a.zz {
    c.type == "Complete" and c.status == "True"
} 

```

### 2. What did you expect to see? (Required)
`True` - I'd expect any valid expression to be viable in `in` part of `any`

### 3. What did you see instead (Required)
```
error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:3:16
  |
3 | b = any c in _a?.zz or [] {
  |                ^ expected one of ["{"] got ?
  |


error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:3:16
  |
3 | b = any c in _a?.zz or [] {
  |                ^ expected one of ["identifier", "literal", "(", "[", "{"] got ?
  |


error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:3:27
  |
3 | b = any c in _a?.zz or [] {
  |                           ^ expected one of ["}"] got {
  |


error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:4:47
  |
4 |     c.type == "Complete" and c.status == "True"
  |                                               ^ expected one of [":", "=", "+="] got newline
  |


error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:5:1
  |
5 | }
  |  expected one of ["identifier", "literal", "(", "[", "{"] got dedent
  |


error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:3:16
  |
3 | b = any c in _a?.zz or [] {
  |                ^ expected one of ["{"] got ?
  |


error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:3:16
  |
3 | b = any c in _a?.zz or [] {
  |                ^ expected one of ["identifier", "literal", "(", "[", "{"] got ?
  |


error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:3:27
  |
3 | b = any c in _a?.zz or [] {
  |                           ^ expected one of ["}"] got {
  |


error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:4:47
  |
4 |     c.type == "Complete" and c.status == "True"
  |                                               ^ expected one of [":", "=", "+="] got newline
  |


error[E1001]: InvalidSyntax
 --> /home/vladimir/zz/z.k:5:1
  |
5 | }
  |  expected one of ["identifier", "literal", "(", "[", "{"] got dedent
  |


error[E2L23]: CompileError
 --> /home/vladimir/zz/z.k:4:5
  |
4 |     c.type == "Complete" and c.status == "True"
  |     ^ name 'c' is not defined
  |


error[E2L23]: CompileError
 --> /home/vladimir/zz/z.k:4:30
  |
4 |     c.type == "Complete" and c.status == "True"
  |                              ^ name 'c' is not defined
  |


error[E2A31]: IllegalAttributeError
 --> /home/vladimir/zz/z.k:4:5
  |
4 |     c.type == "Complete" and c.status == "True"
  |     ^ A attribute must be string type, got 'bool'
  |
```
### 4. What is your KCL components version? (Required)

0.11.1-linux-amd64

--- a/crates/parser/src/parser/expr.rs
+++ b/crates/parser/src/parser/expr.rs
@@ -730,7 +730,7 @@ impl<'a> Parser<'a> {
     /// Syntax:
     /// quant_expr: quant_op [ identifier COMMA ] identifier IN quant_target LEFT_BRACE (expr [IF expr]
     /// | NEWLINE _INDENT simple_expr [IF expr] NEWLINE _DEDENT)? RIGHT_BRACE
-    /// quant_target: string | identifier | list_expr | list_comp | dict_expr | dict_comp
+    /// quant_target: string | identifier | list_expr | list_comp | dict_expr | dict_comp | selector_expr
     /// quant_op: ALL | ANY | FILTER | MAP
     fn parse_quant_expr(&mut self) -> NodeRef<Expr> {
         let token = self.token;
@@ -886,8 +886,38 @@ impl<'a> Parser<'a> {
                     );
                     self.missing_expr()
                 } else {
-                    // identifier
-                    self.parse_identifier_expr()
+                    // identifier - parse identifier and then suffixes like ., ?., []
+                    let lo = self.token;
+                    let mut operand: NodeRef<Expr> = Box::new(Node::new_with_loc(
+                        Expr::Identifier(self.parse_identifier().node),
+                        self.sess.struct_token_loc(lo, self.prev_token),
+                    ));
+
+                    // Parse suffixes: ., ?., [], ?[]
+                    loop {
+                        match self.token.kind {
+                            TokenKind::Dot => {
+                                operand = self.parse_selector_expr(operand, lo);
+                            }
+                            TokenKind::Question => match self.cursor.peek() {
+                                Some(token) => match token.kind {
+                                    TokenKind::Dot => {
+                                        operand = self.parse_selector_expr(operand, lo);
+                                    }
+                                    TokenKind::OpenDelim(DelimToken::Bracket) => {
+                                        operand = self.parse_subscript_expr(operand, lo);
+                                    }
+                                    _ => break,
+                                },
+                                None => break,
+                            },
+                            TokenKind::OpenDelim(DelimToken::Bracket) => {
+                                operand = self.parse_subscript_expr(operand, lo);
+                            }
+                            _ => break,
+                        }
+                    }
+                    operand
                 }
             }
             TokenKind::Literal(lk) => {

--- a/crates/parser/src/tests/expr.rs
+++ b/crates/parser/src/tests/expr.rs
@@ -81,6 +81,11 @@ parse_expr_snapshot!(
     quant_expr_6,
     r####"map i, e in [{k1 = "v1", k2 = "v2"}] { e if i > 0 }"####
 );
+// Test safe navigation operator in quant expressions (issue #2048)
+parse_expr_snapshot!(quant_safe_nav_0, r####"any x in _a?.b { x > 0 }"####);
+parse_expr_snapshot!(quant_safe_nav_1, r####"all y in _a?.b?.c { y < 0 }"####);
+parse_expr_snapshot!(quant_safe_nav_2, r####"filter z in _a?.items { z > 1 }"####);
+parse_expr_snapshot!(quant_safe_nav_3, r####"map w in _a?.items?[0] { w }"####);
 parse_expr_snapshot!(lambda_expr_0, r####"lambda {}"####);
 parse_expr_snapshot!(lambda_expr_1, r####"lambda x {}"####);
 parse_expr_snapshot!(lambda_expr_2, r####"lambda x: int -> int {x}"####);

